lib/kubernetes-deploy/concurrency.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  module Concurrency
+    MAX_THREADS = 8
+
+    def self.split_across_threads(all_work, &block)
+      return if all_work.empty?
+      raise ArgumentError, "Block of work is required" unless block_given?
+
+      slice_size = ((all_work.length + MAX_THREADS - 1) / MAX_THREADS)
```

Comment:
Yeah, I considered `Queue` as well and didn't use it for exactly that reason--it wouldn't be as simple, and the expected benefit would be low given that these pieces of work should be roughly equal.